commands/kafka/cluster/create.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/json2table/jsonpaths"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/jsontabwriter"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/tabheaders"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/pointer"
+	kafka "github.com/ionos-cloud/sdk-go-kafka"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func Create() *core.Command {
+	cmd := core.NewCommand(context.Background(), nil, core.CommandBuilder{
+		Namespace: "kafka",
+		Resource:  "cluster",
+		Verb:      "create",
+		Aliases:   []string{"c", "post"},
+		ShortDesc: "Create a kafka cluster. Wiki: https://docs.ionos.com/cloud/data-analytics/kafka/api-howtos/create-kafka",
+		Example:   "ionosctl kafka cl create --name my-cluster --version 1.0 --size S --location de/txl --datacenter-id DATACENTER_ID --lan-id LAN_ID --broker-addresses 127.0.0.1,127.0.0.2",
+		PreCmdRun: func(c *core.PreCommandConfig) error {
+			if err := core.CheckRequiredFlags(c.Command, c.NS,
+				constants.FlagName, constants.FlagVersion, constants.FlagSize, constants.FlagLocation,
+				constants.FlagDatacenterId, constants.FlagLanId, constants.FlagKafkaBrokerAddresses); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		CmdRun: func(c *core.CommandConfig) error {
+			input := &kafka.Cluster{}
+			if err := setPropertiesFromFlags(c, input); err != nil {
+				return err
+			}
+
+			changeLocation(client.Must().Kafka, viper.GetString(core.GetFlagName(c.NS, constants.FlagLocation)))
+			res, _, err := client.Must().Kafka.ClustersApi.ClustersPost(context.Background()).
+				ClusterCreate(kafka.ClusterCreate{
+					Properties: input,
+				}).Execute()
+			if err != nil {
+				return err
+			}
+
+			return printCluster(c, res)
+		},
+		InitClient: true,
+	})
+
+	cmd.Command.SilenceUsage = true
+	cmd.Command.Flags().SortFlags = false
+	return addClusterCreateFlags(cmd)
+}
+
+func changeLocation(client *kafka.APIClient, location string) {
+	cfg := client.GetConfig()
+	cfg.Servers = kafka.ServerConfigurations{
+		{
+			URL: locationToURL[location],
+		},
+	}
+	return
+}
+
+func addClusterCreateFlags(cmd *core.Command) *core.Command {
+	cmd.AddStringFlag(constants.FlagName, "", "", "The name of the kafka cluster", core.RequiredFlagOption())
+	cmd.AddStringFlag(constants.FlagVersion, "", "", "The version of the kafka cluster", core.RequiredFlagOption())
+	cmd.AddSetFlag(constants.FlagSize, "", "", []string{"XS", "S", "M", "L", "XL"}, "The size of the kafka cluster", core.RequiredFlagOption())
+	_ = cmd.Command.RegisterFlagCompletionFunc(constants.FlagSize, func(c *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return []string{"XS", "S", "M", "L", "XL"}, cobra.ShellCompDirectiveNoFileComp
+	})
+
+	cmd.AddStringFlag(constants.FlagLocation, "", "", "The location of the kafka cluster", core.RequiredFlagOption())
```

Comment:
Could we add this as a persistent flag on the `kafka` command?

Also, some users can't use completions due to the nature of their environment so we should add the possible values to the helptext of the flag

` slices.Collect(maps.Keys(locationToURL))`